api/python/provisioner/commands/check.py

```diff
@@ -609,6 +616,177 @@ def _cluster_status(*, args: str) -> CheckEntry:
 
         return res
 
+    @staticmethod
+    def _network_drivers(*, args: str) -> List[CheckEntry]:
+        """Validates if Network Drivers are proper
+        network_drivers <driver_name> <nodes>
+        """
+
+        res: List[CheckEntry] = list()
+        driver_name = [cfg.NETWORK_DRIVER]
+
+        if cortx_py_utils_import_error:
+            err_res : CheckEntry = CheckEntry(cfg.Checks.NETWORK_DRIVERS.value)
+            err_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                             comment="Package cortx-py-utils not installed")
+            res.append(err_res)
+            return res
+
+        check_value : CheckEntry = CheckEntry(cfg.Checks.NETWORK_DRIVERS.value)
+        try:
+            nodes = Check._get_pillar_data("cluster/node_list")
+            check_res : CheckEntry = CheckEntry(
+                                     cfg.Checks.NETWORK_DRIVERS.value)
+            try:
+                driver_args = driver_name + nodes
+                NetworkV().validate('drivers', driver_args)
+                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
+                                     comment="Network Driver "
+                                     f"Validated: {driver_args}")
+
+            except Exception as exc:
+                check_res.set_fail(checked_target=cfg.ALL_MINIONS,
+                                   comment=str(exc))
+            res.append(check_res)
+
+        except Exception as exc:
+            check_value.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            res.append(check_value)
```

Comment:
I think the nested `try/except` block is not needed there. It will be more easily to read the code and understand what code is expected to be failure when `try/except` block places one-by-one. Also, please consider to move
```python
                check_res.set_passed(checked_target=cfg.ALL_MINIONS,
                                     comment="Network Driver "
                                     f"Validated: {driver_args}")
```
to `else` block of `try/except`. Also, you always return `res` list only with one element. I think `res` is not needed here. So as result you can use just one instance of `CheckEntry` here. Moreover, I found that we can just join these two blocks of `try/except` to the single one because both `except`-blocks treated the same.

So as result I propose to rewrite this method and others(if necessary) to the following form.
```python
@staticmethod
def _network_drivers(*, args: str) -> CheckEntry:
    """
    Validates if Network Drivers are proper
    network_drivers <driver_name> <nodes>
    :param str args: validation params
    :return: CheckEntry with validation result
    """

    driver_name = [cfg.NETWORK_DRIVER]

    check_res: CheckEntry = CheckEntry(cfg.Checks.NETWORK_DRIVERS.value)

    if cortx_py_utils_import_error:
        check_res.set_fail(checked_target=cfg.ALL_MINIONS,
                           comment="Package cortx-py-utils not installed")
        return check_res

    try:
        nodes = Check._get_pillar_data("cluster/node_list")

        driver_args = driver_name + nodes

        NetworkV().validate('drivers', driver_args)
    except Exception as exc:
        check_res.set_fail(checked_target=cfg.ALL_MINIONS,
                           comment=str(exc))
    else:
        check_res.set_passed(checked_target=cfg.ALL_MINIONS,
                             comment="Network Driver "
                                     f"Validated: {driver_args}")

    return check_res
```

It is more simple, takes fewer lines of codes and local variables, easy to read and maintain

Reply:
Yes, since we do not have a need to append the exceptions, this code can be handled in a single `try-except-else` block. Comment addressed.